### DISCUSSION TITLE
Azure App Service and SSH (Git) ?

### DISCUSSION THREAD

**Discussion by ahouziaux** [requarks/wiki#4637] [Help / Questions]
Hey guys,

I recently deployed Wiki JS on Azure. I am now trying to link it to a Github repository using the dedicated storage module. 
I am trying to accomplish that through SSH. 

The issue is that as my Wiki JS is deployed as an Azure App, I am unable to SSH (using a password based authentication) to use the ssh-keygen and then use the generated keys.
I tried to look around the web and found multiple sources about how SSH is supposed to work for web app : the port 2222 is supposed to be exposed, and a specific password is supposed to be used. The sshd_config is also supposed to indicate the port 2222. (Here is the documentation published by MS : https://docs.microsoft.com/en-us/azure/app-service/tutorial-custom-container?pivots=container-linux)

So I just built my own image but... it does not work. It happens that my sshd_config is replaced or just not imported when deploying the new Docker image.

Do any of you successfuly managed to deploy Wiki.JS as an Azure App while making use of the Git storage through SSH ? 
As you may have understood it, I don't really care about being able to SSH into the web app, I just want to be able to use the Git module through SSH and if there is another way to do so, that would be totally fine.

I don't know if it can be helpful but here is the Dockerfile for the Azure App Service version of Node : https://github.com/Azure-App-Service/node/blob/master/10.14/Dockerfile

Any help would be greatly appreciated :)

**Comment by tr3ysmith** [comment]
@ahouziaux I'm not really sure why I didn't think about this, but In the Git setup section on your Wiki, theres an option to specify the private key contents instead of using a path. Just create the key pair on your local machine using WSL, Mac, or Linux and then paste the private key contents into the box labeled "B" See below.

Mine looks to be syncing just fine now!


![image](https://user-images.githubusercontent.com/15764606/140972413-faa64631-a6ff-4a3f-a005-fff4ed83714e.png)
